pkg/managed/services_test.go

```diff
@@ -5,97 +5,221 @@ package managed
+// TestCreateServicesVMIBinding tests the creation of services for a VMI system binding
+// GIVEN a cluster which does not have any services
+//  WHEN I call CreateServices
+//  THEN there should be a Node Exporter service created
 func TestCreateServicesVMIBinding(t *testing.T) {
+	assert := assert.New(t)
+
 	mbPairSystemBinding := testutil.GetModelBindingPair()
 	mbPairSystemBinding.Model.Name = constants.VmiSystemBindingName
 	mbPairSystemBinding.Binding.Name = constants.VmiSystemBindingName
 
 	// Model/binding for VMI system binding - Node Exporter service is created.
 	managedConnections := testutil.GetManagedClusterConnections()
 	err := CreateServices(mbPairSystemBinding, managedConnections)
-	if err != nil {
-		t.Fatal(fmt.Sprintf("can't create service: %v", err))
-	}
+	assert.Nil(err, "got an error from CreateServices: %v", err)
 
 	for clusterName := range mbPairSystemBinding.ManagedClusters {
 		managedClusterConnection := managedConnections[clusterName]
 		existingService, err := managedClusterConnection.KubeClient.CoreV1().Services(constants.MonitoringNamespace).List(context.TODO(), metav1.ListOptions{})
-		if err != nil {
-			t.Fatal(fmt.Sprintf("can't list service for cluster %s: %v", err, clusterName))
-		}
-		assertCreateService(t, existingService, clusterName)
+		assert.Nil(err, "got an error listing services: %v", err)
+		assertCreateServiceVMI(t, existingService, clusterName)
 
 		// Update the port value so when we call CreateServices again the update code is executed.
 		existingService.Items[0].Spec.Ports[0].Port = 9200
 		_, err = managedClusterConnection.KubeClient.CoreV1().Services(constants.MonitoringNamespace).Update(context.TODO(), &existingService.Items[0], metav1.UpdateOptions{})
-		if err != nil {
-			t.Fatal(fmt.Sprintf("can't update service for cluster %s: %v", err, clusterName))
-		}
+		assert.Nil(err, "got an error updating services: %v", err)
 	}
 
 	// Model/binding for VMI system binding - service already exist
 	err = CreateServices(mbPairSystemBinding, managedConnections)
-	if err != nil {
-		t.Fatal(fmt.Sprintf("can't create service: %v", err))
-	}
+	assert.Nil(err, "got an error from CreateServices: %v", err)
 
 	for clusterName := range mbPairSystemBinding.ManagedClusters {
 		managedClusterConnection := managedConnections[clusterName]
 		existingService, err := managedClusterConnection.KubeClient.CoreV1().Services(constants.MonitoringNamespace).List(context.TODO(), metav1.ListOptions{})
-		if err != nil {
-			t.Fatal(fmt.Sprintf("can't list service for cluster %s: %v", err, clusterName))
-		}
-		assertCreateService(t, existingService, clusterName)
+		assert.Nil(err, "got an error listing services: %v", err)
+		assertCreateServiceVMI(t, existingService, clusterName)
+	}
+}
+
+// TestDeleteServices tests that an existing service is properly deleted.
+// GIVEN a cluster which has an existing generic component service (test-generic)
+//  WHEN I call CreateServices
```

Comment:
should this be when I call `DeleteServices`?

Reply:
oops. cut-and-paste error. fixed.